### DISCUSSION TITLE
Apply Spatie Media Library to existing Folder Structure

### DISCUSSION THREAD

**Discussion by andreas-onlineschmiede** [spatie/laravel-medialibrary#2472] [Q&A]
Hello everyone!

I am using the laravel-medialibrary 9.0 with laravel 8.12 and PHP 7.4:

I want to use all the features of the image library, but I already have a large application with a folder structure of images in /storage.

Normally, if I call 
```
$yourModel->addMedia($pathToFile)
```
it will remove the file from its original location and put it in its own path.

For example, I have a file under /storage/4/9/49.jpg and I want to have it associated to a model, without removing it from this folder or creating a new file, I just want to link it, so it has an entry in the medias table and I can access it with
```
$mediaItems = $yourModel->getMedia();
```

Is there any possibility, to link all existing images to my models?
Basically I want to manually fill the media table.

Thank you a lot for your tips!
